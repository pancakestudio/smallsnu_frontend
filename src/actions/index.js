import * as types from './ActionTypes';

export function buildingClick(e){
  return{
    type: types.BUILDING_CLICK,
    e
  };
}
