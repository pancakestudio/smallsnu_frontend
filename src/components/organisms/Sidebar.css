#sidebar {
  border-radius: 5px;
  height: calc(100vh - 85px);
  min-width: 250px;
  max-width: 250px;
  padding: 10px;
  position: fixed;
  top: 65px;
  margin: 10px;
  left: -300px;
  z-index: 9999;
  transition: all 0.3s;
  overflow-y: scroll;
  box-shadow: 0 0 5px 1.5px rgba(0,0,0,0.4);
}

#sidebar.active {
  left: 0;
}

@media(max-width: 575px){
  #sidebar {
    top: 110px;
    height: calc(100vh - 130px);
  }
}
