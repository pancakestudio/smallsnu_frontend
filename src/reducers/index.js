import { combineReducers } from 'redux';
import buildingButton from './buildingButton';

const reducers = combineReducers({
  buildingButton
});

export default reducers;
