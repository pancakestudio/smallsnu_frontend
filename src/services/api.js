import axios from 'axios'

axios.defaults.baseURL = 'http://127.0.0.1:8000/'

export function getBuildingInfo(bldgNo){
  return axios.get(`building/${bldgNo}`)
    .catch(error => { return {error: error}})
}

export function getRestaurantInfo(){
  return axios.get(`restaurant/`)
    .catch(error => {return {error: error}})
}

export function getSeminarInfo(){
  return axios.get('seminar/')
    .catch(error => { return {error:error}})
}
