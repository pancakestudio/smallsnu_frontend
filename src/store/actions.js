import * as types from './actionTypes'

/******** Map ********/

export function requestBuilding(bldgNo){
  return{
    type: types.REQUEST_BUILDING,
    bldgNo
  };
}

export function getBuildingSuccess(krName, bldgNo, info, restaurants, seminars, lectures, posts){
  return{
    type: types.GET_BUILDING_SUCCESS,
    krName,
    bldgNo,
    info,
    restaurants,
    seminars,
    lectures,
    posts,
  }
}

export function getBuildingFailure(error){
  return{
    type: types.GET_BUILDING_FAILURE,
    error
  }
}

export function zoomChanged(zoomLevel){
  return {
    type: types.ZOOM_CHANGED,
    zoomLevel
  }
}

export function hideMarkers(){
  return {
    type: types.HIDE_MARKERS
  }
}

/******** SideBar ********/

export function toggleSidebar(){
  return {
    type: types.TOGGLE_SIDE_BAR
  }
}

export function toggleShuttleMenuMarker(){
  return{
    type: types.TOGGLE_SHUTTLE_MENU_MARKER
  }
}

/******** SearchBar ********/

export function search(bldgNo, bldgPos){
  return{
    type: types.SEARCH,
    bldgNo,
    bldgPos
  }
}

/******** Shuttle ********/
export function requestAllShuttles(){
  return{
    type: types.REQUEST_ALL_SHUTTLE
  }
}

export function getAllShuttleSuccess(stations){
  return{
    type: types.GET_ALL_SHUTTLE_SUCCESS,
    stations
  }
}

export function toggleShuttleMarker(){
  return{
    type: types.TOGGLE_SHUTTLE_MARKER
  }
}

export function requestAllRevShuttles(){
  return{
    type: types.REQUEST_ALL_REV_SHUTTLE
  }
}

export function getAllRevShuttleSuccess(stations){
  return{
    type: types.GET_ALL_REV_SHUTTLE_SUCCESS,
    stations
  }
}

export function toggleRevShuttleMarker(){
  return{
    type: types.TOGGLE_REV_SHUTTLE_MARKER
  }
}

/******** Post ********/

export function requestBoard(bldgNo){
  return {
    type: types.REQUEST_BOARD,
    bldgNo
  }
}

export function getBoardSuccess(posts, bldgNo){
  return {
    type: types.GET_BOARD_SUCCESS,
    posts,
    bldgNo
  }
}

export function getBoardFailure(error){
  return {
    type: types.GET_BOARD_FAILURE,
    error
  }
}

export function requestPost(id) {
  return {
    type: types.REQUEST_POST,
    id
  }
}

export function getPostSuccess(post){
  return {
    type: types.GET_POST_SUCCESS,
    post
  }
}

export function getPostFailure(error){
  return {
    type: types.GET_POST_FAILURE,
    error
  }
}

export function savePost(post, bldgNo){
  return{
    type: types.SAVE_POST,
    post,
    bldgNo
  }
}

export function savePostSuccess(bldgNo){
  return {
    type: types.SAVE_POST_SUCCESS,
    bldgNo
  }
}

export function savePostFailure(error){
  return {
    type: types.SAVE_POST_FAILURE,
    error
  }
}

export function editPost(post, bldgNo){
  return{
    type: types.EDIT_POST,
    post,
    bldgNo,
  }
}

export function editPostSuccess(postId){
  return {
    type: types.EDIT_POST_SUCCESS,
    postId
  }
}

export function editPostFailure(error){
  return {
    type: types.EDIT_POST_FAILURE,
    error
  }
}

export function editPostFlag(){
  return{
    type: types.EDIT_POST_FLAG,
  }
}

export function deletePost(post, bldgNo, password){
  return{
    type: types.DELETE_POST,
    post,
    bldgNo,
    password
  }
}

export function deletePostSuccess(bldgNo){
  return {
    type: types.DELETE_POST_SUCCESS,
    bldgNo
  }
}

export function deletePostFailure(error){
  return {
    type: types.DELETE_POST_FAILURE,
    error
  }
}

export function showWritePost(bldgNo) {
  return{
    type: types.SHOW_WRITE_POST,
    bldgNo
  };
}

export function hideWritePost() {
  return {
    type: types.HIDE_WRITE_POST,
  }
}

export function showPasswordCheck(target){
  return{
    type: types.SHOW_PASSWORD_CHECK,
    target
  }
}

export function hidePasswordCheck(){
  return{
    type: types.HIDE_PASSWORD_CHECK
  }
}

export function postLike(postId) {
  return {
    type: types.POST_LIKE,
    postId
  }
}

export function postLikeSuccess(postId) {
  return {
    type: types.POST_LIKE_SUCCESS,
    postId
  }
}

export function postLikeFailure(error) {
  return {
    type: types.POST_LIKE_FAILURE,
    error
  }
}

export function saveComment(comment, postId){
  return {
    type: types.SAVE_COMMENT,
    comment,
    postId
  }
}

export function saveCommentSuccess(postId){
  return {
    type: types.SAVE_COMMENT_SUCCESS,
    postId
  }
}

export function saveCommentFailure(error){
  return {
    type: types.SAVE_COMMENT_FAILURE,
    error
  }
}

export function showEditComment(comment){
  return {
    type: types.SHOW_EDIT_COMMENT,
    comment
  }
}

export function hideEditComment(){
  return {
    type: types.HIDE_EDIT_COMMENT
  }
}

export function editComment(comment, postId){
  return {
    type: types.EDIT_COMMENT,
    comment,
    postId
  }
}

export function editCommentSuccess(postId){
  return {
    type: types.EDIT_COMMENT_SUCCESS,
    postId
  }
}

export function editCommentFailure(error){
  return {
    type: types.EDIT_COMMENT_FAILURE,
    error
  }
}

export function deleteComment(comment, postId, password){
  return {
    type: types.DELETE_COMMENT,
    comment,
    postId,
    password
  }
}

export function deleteCommentSuccess(msg){
  return {
    type: types.DELETE_COMMENT_SUCCESS,
    msg
  }
}

export function deleteCommentFailure(error){
  return {
    type: types.DELETE_COMMENT_FAILURE,
    error
  }
}

export function commentLike(commentId, postId) {
  return {
    type: types.COMMENT_LIKE,
    commentId,
    postId
  }
}

export function commentLikeSuccess(postId) {
  return {
    type: types.COMMENT_LIKE_SUCCESS,
    postId
  }
}

export function commentLikeFailure(error) {
  return {
    type: types.COMMENT_LIKE_FAILURE,
    error
  }
}

export function changeBoardPage(page){
  return {
    type: types.CHANGE_BOARD_PAGE,
    page
  }
}

export function wrongPassword(){
  return {
    type: types.WRONG_PASSWORD,
  }
}

/******** Restaurant ********/

export function toggleResMarker(){
  return{
    type: types.TOGGLE_RES_MARKER,
  }
}

export function requestRestaurant(id){
  return{
    type: types.REQUEST_RESTAURANT,
    id
  }
}

export function getRestaurantSuccess(restaurant){
  return{
    type : types.GET_RESTAURANT_SUCCESS,
    restaurant,
  }
}

export function getRestaurantFailure(error){
  return{
    type: types.GET_RESTAURANT_FAILURE,
    error
  }
}

export function requestAllRestaurants(){
  return{
    type: types.REQUEST_ALL_RESTAURANTS,
  }
}


export function getAllRestaurantsSuccess(restaurants){
  return{
    type : types.GET_ALL_RESTAURANTS_SUCCESS,
    restaurants,
  }
}

export function getAllRestaurantsFailure(error){
  return{
    type: types.GET_ALL_RESTAURANTS_FAILURE,
    error
  }
}

/******** Seminar ********/

export function toggleSemiMarker(){
  return{
    type: types.TOGGLE_SEMI_MARKER,
  }
}

export function changeSeminarPage(page){
  return {
    type: types.CHANGE_SEMINAR_PAGE,
    page
  }
}

export function requestSeminar(id){
  return {
    type: types.REQUEST_SEMINAR,
    id
  }
}

export function getSeminarSuccess(seminar){
  return {
    type: types.GET_SEMINAR_SUCCESS,
    seminar
  }
}

export function getSeminarFailure(error){
  return {
    type: types.GET_SEMINAR_FAILURE,
    error
  }
}

export function requestBldgSeminars(bldgNo){
  return {
    type: types.REQUEST_BLDG_SEMINARS,
    bldgNo
  }
}

export function getBldgSeminarsSuccess(seminars, bldgNo){
  return {
    type: types.GET_BLDG_SEMINARS_SUCCESS,
    seminars,
    bldgNo
  }
}

export function getBldgSeminarsFailure(error){
  return {
    type: types.GET_BLDG_SEMINARS_FAILURE,
    error
  }
}

export function requestAllSeminars(){
  return {
    type: types.REQUEST_ALL_SEMINARS,
  }
}

export function getAllSeminarsSuccess(seminars){
  return {
    type: types.GET_ALL_SEMINARS_SUCCESS,
    seminars
  }
}

export function getAllSeminarsFailure(error){
  return {
    type: types.GET_ALL_SEMINARS_FAILURE,
    error
  }
}

/******** Bank ********/

export function toggleBankMarker(){
  return{
    type: types.TOGGLE_BANK_MARKER,
  }
}

export function requestAllBanks(){
  return{
    type: types.REQUEST_ALL_BANKS,
  }
}

export function getAllBanksSuccess(banks){
  return{
    type: types.GET_ALL_BANKS_SUCCESS,
    banks,
  }
}

export function getAllBanksFailure(error){
  return{
    type: types.GET_ALL_BANKS_FAILURE,
    error,
  }
}

export function requestBank(id){
  return{
    type: types.REQUEST_BANK,
    id
  }
}

export function getBankSuccess(bank){
  return{
    type: types.GET_BANK_SUCCESS,
    bank
  }
}

export function getBankFailure(error){
  return{
    type: types.GET_BANK_FAILURE,
    error
  }
}


/******** ATM ********/

export function toggleATMMarker(){
  return{
    type: types.TOGGLE_ATM_MARKER,
  }
}

export function requestAllATMs(){
  return{
    type: types.REQUEST_ALL_ATMS,
  }
}

export function getAllATMsSuccess(atms){
  return{
    type: types.GET_ALL_ATMS_SUCCESS,
    atms,
  }
}

export function getAllATMsFailure(error){
  return{
    type: types.GET_ALL_ATMS_FAILURE,
    error,
  }
}

export function requestATM(id){
  return{
    type: types.REQUEST_ATM,
    id
  }
}

export function getATMSuccess(atm){
  return{
    type: types.GET_ATM_SUCCESS,
    atm
  }
}

export function getATMFailure(error){
  return{
    type: types.GET_ATM_FAILURE,
    error
  }
}

/******** Cafe ********/

export function toggleCafeMarker(){
  return{
    type: types.TOGGLE_CAFE_MARKER,
  }
}

export function requestAllCafes(){
  return{
    type: types.REQUEST_ALL_CAFES,
  }
}

export function getAllCafesSuccess(cafes){
  return{
    type: types.GET_ALL_CAFES_SUCCESS,
    cafes,
  }
}

export function getAllCafesFailure(error){
  return{
    type: types.GET_ALL_CAFES_FAILURE,
    error,
  }
}

export function requestCafe(id){
  return{
    type: types.REQUEST_CAFE,
    id
  }
}

export function getCafeSuccess(cafe){
  return{
    type: types.GET_CAFE_SUCCESS,
    cafe
  }
}

export function getCafeFailure(error){
  return{
    type: types.GET_CAFE_FAILURE,
    error
  }
}

/******** Conv ********/

export function toggleConvMarker(){
  return{
    type: types.TOGGLE_CONV_MARKER,
  }
}

export function requestAllConves(){
  return{
    type: types.REQUEST_ALL_CONVES,
  }
}

export function getAllConvesSuccess(conves){
  return{
    type: types.GET_ALL_CONVES_SUCCESS,
    conves,
  }
}

export function getAllConvesFailure(error){
  return{
    type: types.GET_ALL_CONVES_FAILURE,
    error,
  }
}

export function requestConv(id){
  return{
    type: types.REQUEST_CONV,
    id
  }
}

export function getConvSuccess(conv){
  return{
    type: types.GET_CONV_SUCCESS,
    conv
  }
}

export function getConvFailure(error){
  return{
    type: types.GET_CONV_FAILURE,
    error
  }
}


/******** Main ********/

export function refreshData(){
  return {
    type: types.REFRESH_DATA
  }
}
