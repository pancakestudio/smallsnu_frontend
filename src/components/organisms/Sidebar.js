import React from 'react'
import { Container, Row, Col, Card, Form, FormControl, Nav, ListGroup, Button } from 'react-bootstrap'
import { FaAngleRight, FaAngleLeft, FaSearch } from 'react-icons/fa'
import { getBldgCoord, getBldgNo, getNearestSpot } from '../../utils/Functions'
import './Sidebar.css'

export const Sidebar = ({show, shuttleShow, pathFind, onBack, onPathFindClick, onResClick,
  source, destination, path, onSearchSrc, onSearchDest, onPickSrc, onPickDest, onFind,
  onSemiClick, onCafeClick, onConvClick, onBankClick, onATMClick,
  onShuttleMenuClick, onShuttleClick, onRevShuttleClick,
  onSchoolShuttleClick, onMidLibShuttleClick, onMidShuttleClick, hide}) => {
  let src, dest
  const handleBack = () => {
    onBack()
  }
  const srcSearch = (e) => {
    e.preventDefault()
    let srcNo = src
    let coord = getBldgCoord(srcNo)
    if(!coord){
      srcNo = srcNo.slice(0,-1)
      coord = getBldgCoord(srcNo)
    }
    if(coord){
      const srcPos = {lat: coord[0], lng: coord[1]}
      onSearchSrc(srcNo, srcPos)
      e.target.reset()
    } else {
      alert("해당 건물번호를 가진 건물이 없습니다.")
      e.target.reset()
    }
  }
  const handleSrcChange = (e) => {
    src = e.target.value
  }
  const handlePickSrc = () => {
    onPickSrc()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const destSearch = (e) => {
    e.preventDefault()
    let destNo = dest
    let coord = getBldgCoord(destNo)
    if(!coord){
      destNo = destNo.slice(0,-1)
      coord = getBldgCoord(destNo)
    }
    if(coord){
      const destPos = {lat: coord[0], lng: coord[1]}
      onSearchDest(destNo, destPos)
      e.target.reset()
    } else {
      alert("해당 건물번호를 가진 건물이 없습니다.")
      e.target.reset()
    }
  }
  const handlePickDest = () => {
    onPickDest()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleDestChange = (e) => {
    dest = e.target.value
  }
  const handleFind = () => {
    if(source.pos && destination.pos){
      onFind(getNearestSpot(source.pos), getNearestSpot(destination.pos))
      if(window.innerWidth<=576){
        hide()
      }
    } else {
      alert("출발지와 목적지를 설정해주세요.")
    }
  }

  const handlePathFindClick = () => {
    onPathFindClick()
  }
  const handleSemiClick = () => {
    onSemiClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleShuttleClick = () => {
    onShuttleClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleRevShuttleClick = () => {
    onRevShuttleClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleSchoolShuttleClick = () => {
    onSchoolShuttleClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleMidLibShuttleClick = () => {
    onMidLibShuttleClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleMidShuttleClick = () => {
    onMidShuttleClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleShuttleMenuClick = () => {
    onShuttleMenuClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleResClick = () => {
    onResClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleCafeClick = () => {
    onCafeClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleConvClick = () => {
    onConvClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleBankClick = () => {
    onBankClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  const handleATMClick = () => {
    onATMClick()
    if(window.innerWidth<=576){
      hide()
    }
  }
  let sidebar, className = "flex-column bg-light"
  let srcDefault, destDefault, srcDestInfo, pathInfo
  if(show){
    className = "active " + className
  }

  if(source && Object.keys(source).length!==0){
    if(source.pick && source.pos){
      srcDefault = getBldgNo(source.pos)+"동"
      if(srcDefault==="0동")
        srcDefault = getNearestSpot(source.pos).code
    } else if(source.bldgNo){
      srcDefault = source.bldgNo+"동"
    }
  }

  if(destination && Object.keys(destination).length!==0){
    if(destination.pick && destination.pos){
      destDefault = getBldgNo(destination.pos)+"동"
      if(destDefault==="0동")
        destDefault = getNearestSpot(destination.pos).code
    } else if(destination.bldgNo){
      destDefault = destination.bldgNo+"동"
    }
  }

  if(path && Object.keys(path).length>1){
    srcDestInfo = (
      <Row className="infoWrapper">
        <Card className="srcDestCard">
          <Card.Header as="h5">경로 정보</Card.Header>
          <Card.Body>
            <Card.Text>
              {path.src} <FaAngleRight/> {path.dest}
            </Card.Text>
          </Card.Body>
        </Card>
      </Row>
    )
    pathInfo = (
      <Row className="infoWrapper">
        <Card body className="pathCard">
          <Card.Text>
            약 {path.time.slice(0,3)+"분"} | {path.length.slice(0,3)+"m"}
          </Card.Text>
        </Card>
      </Row>
    )
  }

  if(pathFind){
    sidebar = (
      <Nav id="pathFind" className={className}>
        <Container fluid>
          <Row className="wrapper">
            <Col xs={11} md={11} className="pick">
              <Row className="formWrapper">
                <Form className="searchForm" onSubmit={srcSearch} inline>
                  <Col xs={10} md={10} className="inputCol">
                    <FormControl
                      onChange = {handleSrcChange}
                      type = "text"
                      placeholder = "출발(건물 번호)"
                      defaultValue = {srcDefault}
                      className = "textInput"
                    />
                  </Col>
                  <Col xs={2} md={2} className="buttonCol">
                    <Button variant="outline-secondary" type="submit">
                      <FaSearch/>
                    </Button>
                  </Col>
                </Form>
              </Row>
              <Button className="srcMarker" onClick={handlePickSrc}>출발 핀 지정</Button>
              <Row className="formWrapper">
                <Form className="searchForm" onSubmit={destSearch} inline>
                  <Col xs={10} md={10} className="inputCol">
                    <FormControl
                      onChange = {handleDestChange}
                      type = "text"
                      placeholder = "도착(건물 번호)"
                      defaultValue = {destDefault}
                      className = "textInput"
                    />
                  </Col>
                  <Col xs={2} md={2} className="buttonCol">
                    <Button variant="outline-secondary" type="submit"><FaSearch/></Button>
                  </Col>
                </Form>
              </Row>
              <Button className="destMarker" onClick={handlePickDest}>도착 핀 지정</Button>
              <Row className="findWrapper">
                <Button variant="secondary" className="find" onClick={handleFind}>길찾기</Button>
              </Row>
              { srcDestInfo }
              { pathInfo }
            </Col>
            <Button className="back" onClick={handleBack}><FaAngleLeft /></Button>
          </Row>
        </Container>
      </Nav>
    )
  } else if(shuttleShow){
    sidebar = (
      <Nav id="sidebar" className={className}>
        <ListGroup variant="flush">
          <ListGroup.Item className="bg-light" action active={false} onClick={handleShuttleClick}>교내 순환 셔틀</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleRevShuttleClick}>역 순환 셔틀</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleSchoolShuttleClick}>통학 셔틀</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleMidLibShuttleClick}>야간 도서관 셔틀</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleMidShuttleClick}>심야 셔틀</ListGroup.Item>
        </ListGroup>
      </Nav>
    )
  } else {
    sidebar = (
      <Nav id="sidebar" className={className}>
        <ListGroup variant="flush">
          <ListGroup.Item className="bg-light" action active={false} onClick={handlePathFindClick}>길찾기</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleShuttleMenuClick}>셔틀 버스</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleSemiClick}>세미나</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleResClick}>식당</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleCafeClick}>카페</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleConvClick}>편의점</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleBankClick}>은행</ListGroup.Item>
          <ListGroup.Item className="bg-light" action active={false} onClick={handleATMClick}>ATM</ListGroup.Item>
        </ListGroup>
      </Nav>
    )
  }

  return (
    sidebar
  )
}
