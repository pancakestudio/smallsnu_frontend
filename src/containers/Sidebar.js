import { connect } from 'react-redux'
import { Sidebar } from '../components/organisms/Sidebar'
import { sideResClick } from '../store/actions'
import { sideSeminarClick } from '../store/actions'

const mapDispatchToProps = (dispatch) => {
  return {
    onResClick: () => {
      dispatch(sideResClick())
    },
    onSemiClick: () => {
      dispatch((sideSeminarClick()))
    }
  }
}

export default connect(null, mapDispatchToProps)(Sidebar)
