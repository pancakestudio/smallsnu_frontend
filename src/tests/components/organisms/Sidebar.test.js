import React from 'react';
import createSagaMiddleware from 'redux-saga'
import configureStore from 'redux-mock-store'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { applyMiddleware, createStore } from 'redux'
import { Sidebar } from '../../../components/organisms/Sidebar'
import ConnectedSidebar from '../../../containers/Sidebar'
import * as actions from '../../../store/actions'
import rootSaga from '../../../store/sagas'
import reducers from '../../../store/reducers'
import * as api from '../../../services/api'

describe('Sidebar', ()=>{
  let component

  it('renders correctly', ()=>{
    component = shallow(<Sidebar/>)
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has correct menu lists', ()=>{
    expect(component.find('Bootstrap(ListGroupItem)').at(0).text()).toBe('길찾기')
    expect(component.find('Bootstrap(ListGroupItem)').at(1).text()).toBe('셔틀 버스')
    expect(component.find('Bootstrap(ListGroupItem)').at(2).text()).toBe('강의실 예약')
    expect(component.find('Bootstrap(ListGroupItem)').at(3).text()).toBe('세미나')
    expect(component.find('Bootstrap(ListGroupItem)').at(4).text()).toBe('식당')
    expect(component.find('Bootstrap(ListGroupItem)').at(5).text()).toBe('카페')
    expect(component.find('Bootstrap(ListGroupItem)').at(6).text()).toBe('편의점')
    expect(component.find('Bootstrap(ListGroupItem)').at(7).text()).toBe('은행')
    expect(component.find('Bootstrap(ListGroupItem)').at(8).text()).toBe('ATM')
  })

  it('changes className properly', ()=>{
    component.setProps({show: true})
    expect(component.find('.active').exists()).toBe(true)
  })
})

describe('ConnectedSidebar', ()=>{
  const seminar = {
    "title": "[Seminar] Autonomous Driving: Simulation and Navigation",
    "talker": "Dinesh ManochaDepartment of Computer Science and Electrical & Computer Engineering, University of Maryland at College Park",
    "where": "302동 309-1호",
    "time": "2019년 3월 28일 목요일 PM 1:30 - 2019년 3월 28일 목요일 PM 2:30",
    "link": "https://cse.snu.ac.kr/node/36950"
  }
  const restaurant = {
    id: 1,
    kr_name: '식당',
    building: {
      spot: {
        latitude: 37.469,
        longitude: 126.962
      }
    }
  }
  const baseURL = 'http://127.0.0.1:8000'
  const semiRequest = jest
    .spyOn(api, 'getAllSeminarsInfo')
    .mockImplementationOnce(()=>Promise.resolve({
      data: [seminar]
    }))
    .mockImplementationOnce(()=>Promise.resolve({
      data: [seminar]
    }))
    .mockImplementationOnce(()=>Promise.resolve({
      error: "error"
    }))
  const resRequest = jest
    .spyOn(api, 'getAllRestaurantsInfo')
    .mockImplementationOnce(()=>Promise.resolve({
      data: [restaurant]
    }))
    .mockImplementationOnce(()=>Promise.resolve({
      data: [restaurant]
    }))
    .mockImplementationOnce(()=>Promise.resolve({
      error: "error"
    }))
  let store, component

  it('renders correctly', ()=>{
    const sagaMiddleware = createSagaMiddleware()
    const mockStore = configureStore([sagaMiddleware])
    store = mockStore()
    sagaMiddleware.run(rootSaga)
    component = mount(<Provider store={store}><ConnectedSidebar/></Provider>)
  })

  it('matches snapshot', ()=>{
    expect(component).toMatchSnapshot()
  })

  it('has correct menu lists', ()=>{
    expect(component.find('Bootstrap(ListGroupItem)').at(0).text()).toBe('길찾기')
    expect(component.find('Bootstrap(ListGroupItem)').at(1).text()).toBe('셔틀 버스')
    expect(component.find('Bootstrap(ListGroupItem)').at(2).text()).toBe('강의실 예약')
    expect(component.find('Bootstrap(ListGroupItem)').at(3).text()).toBe('세미나')
    expect(component.find('Bootstrap(ListGroupItem)').at(4).text()).toBe('식당')
    expect(component.find('Bootstrap(ListGroupItem)').at(5).text()).toBe('카페')
    expect(component.find('Bootstrap(ListGroupItem)').at(6).text()).toBe('편의점')
    expect(component.find('Bootstrap(ListGroupItem)').at(7).text()).toBe('은행')
    expect(component.find('Bootstrap(ListGroupItem)').at(8).text()).toBe('ATM')
  })

  it('dispatches reqeustAllSeminars and toggleSemiMarker action', ()=>{
    component.find('Bootstrap(ListGroupItem)').at(3).prop('onClick')()
    expect(store.getActions()[0]).toEqual(actions.requestAllSeminars())
    expect(store.getActions()[1]).toEqual(actions.toggleSemiMarker())
  })

  it('handleRequestAllSemiInfo dispatches proper actions', ()=>{
    expect(store.getActions()[2]).toEqual(actions.getAllSeminarsSuccess([seminar]))
  })

  it('dispatches requestAllRestaurants and toggleResMarker action', ()=>{
    component.find('Bootstrap(ListGroupItem)').at(4).prop('onClick')()
    expect(store.getActions()[3]).toEqual(actions.requestAllRestaurants())
    expect(store.getActions()[4]).toEqual(actions.toggleResMarker())
  })

  it('handleRequestAllSemiInfo dispatches proper actions', async ()=>{
    expect(store.getActions()[5]).toEqual(actions.getAllRestaurantsSuccess([restaurant]))
  })

  it('dispatches requestAllCafes and toggleCafeMarker action', ()=>{
    component.find('Bootstrap(ListGroupItem)').at(5).prop('onClick')()
    expect(store.getActions()[6]).toEqual(actions.requestAllCafes())
    expect(store.getActions()[7]).toEqual(actions.toggleCafeMarker())
  })

  it('dispatches requestAllConves and toggleConvMarker action', ()=>{
    component.find('Bootstrap(ListGroupItem)').at(6).prop('onClick')()
    expect(store.getActions()[8]).toEqual(actions.requestAllConves())
    expect(store.getActions()[9]).toEqual(actions.toggleConvMarker())
  })

  it('dispatches requestAllBanks and toggleBankMarker action', ()=>{
    component.find('Bootstrap(ListGroupItem)').at(7).prop('onClick')()
    expect(store.getActions()[10]).toEqual(actions.requestAllBanks())
    expect(store.getActions()[11]).toEqual(actions.toggleBankMarker())
  })

  it('dispatches requestAllATMs and toggleATMMarker action', ()=>{
    component.find('Bootstrap(ListGroupItem)').at(8).prop('onClick')()
    expect(store.getActions()[12]).toEqual(actions.requestAllATMs())
    expect(store.getActions()[13]).toEqual(actions.toggleATMMarker())
  })

  it('reducers', async ()=>{
    const sagaMiddleware = createSagaMiddleware()
    store = createStore(reducers, applyMiddleware(sagaMiddleware))
    sagaMiddleware.run(rootSaga)

    store.dispatch(actions.requestAllSeminars())
    store.dispatch(actions.toggleSemiMarker())
    expect(store.getState().showSemiMarkers).toEqual(true)
    await Promise.resolve()
    expect(store.getState().allSeminars).toEqual([seminar])

    store.dispatch(actions.requestAllSeminars())
    await Promise.resolve()
    expect(store.getState().showSemiMarkers).toEqual(false)
    expect(store.getState().error).toBe('세미나 정보를 받아오지 못했습니다.')

    store.dispatch(actions.requestAllRestaurants())
    store.dispatch(actions.toggleResMarker())
    expect(store.getState().showResMarkers).toEqual(true)
    await Promise.resolve()
    expect(store.getState().allRestaurants).toEqual([restaurant])

    store.dispatch(actions.requestAllRestaurants())
    await Promise.resolve()
    expect(store.getState().showResMarkers).toEqual(false)
    expect(store.getState().error).toBe('식당 정보를 받아오지 못했습니다.')
  })
})
