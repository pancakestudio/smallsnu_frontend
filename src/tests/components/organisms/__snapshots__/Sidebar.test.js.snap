// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedSidebar matches snapshot 1`] = `ReactWrapper {}`;

exports[`Sidebar matches snapshot 1`] = `ShallowWrapper {}`;
